spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/utils/RewriteRowLevelOperationHelper.scala

```diff
@@ -54,12 +54,12 @@ trait RewriteRowLevelOperationHelper extends PredicateHelper with Logging {
       table: Table,
       tableAttrs: Seq[AttributeReference],
       mergeBuilder: MergeBuilder,
-      cond: Expression,
+      cond: Option[Expression] = None,
       matchingRowsPlanBuilder: DataSourceV2ScanRelation => LogicalPlan): LogicalPlan = {
 
     val scanBuilder = mergeBuilder.asScanBuilder
 
-    pushFilters(scanBuilder, cond, tableAttrs)
+    cond.map(pushFilters(scanBuilder, _, tableAttrs))
```

Comment:
I don't think this change is needed because Anton's update already extracts the correct filters from `cond`.

Reply:
@rdblue ok.. i will change it for now. We will discuss the deterministic thing later.